test/UnitTests/TestFramework.UnitTests/Attributes/DataRowAttributeTests.cs

```diff
@@ -84,12 +100,68 @@ public void GetDisplayNameShouldReturnSpecifiedDisplayName()
             DisplayName = "DataRowTestWithDisplayName",
         };
 
-        _dummyTestClass = new DummyTestClass();
-        _testMethodInfo = _dummyTestClass.GetType().GetTypeInfo().GetDeclaredMethod("DataRowTestMethod");
+        var dummyTestClass = new DummyTestClass();
+        var testMethodInfo = dummyTestClass.GetType().GetTypeInfo().GetDeclaredMethod("DataRowTestMethod");
 
         var data = new string[] { "First", "Second", null };
 
-        var displayName = dataRowAttribute.GetDisplayName(_testMethodInfo, data);
+        var displayName = dataRowAttribute.GetDisplayName(testMethodInfo, data);
         Verify(displayName == "DataRowTestWithDisplayName");
     }
+
+    public void GetDisplayNameForArrayOfOneItem()
+    {
+        // Arrange
+        var dataRow = new DataRowAttribute(new[] { "a" });
+        var methodInfoMock = new Mock<MethodInfo>();
+        methodInfoMock.SetupGet(x => x.Name).Returns("MyMethod");
+
+        // Act
+        var displayName = dataRow.GetDisplayName(methodInfoMock.Object, dataRow.Data);
+
+        // Assert
+        Verify(displayName == "MyMethod (a)");
+    }
+
+    public void GetDisplayNameForArrayOfMultipleItems()
+    {
+        // Arrange
+        var dataRow = new DataRowAttribute(new[] { "a", "b", "c" });
+        var methodInfoMock = new Mock<MethodInfo>();
+        methodInfoMock.SetupGet(x => x.Name).Returns("MyMethod");
+
+        // Act
+        var displayName = dataRow.GetDisplayName(methodInfoMock.Object, dataRow.Data);
+
+        // Assert
+        Verify(displayName == "MyMethod (a,b,c)");
+    }
+
+    public void GetDisplayNameForMultipleArraysOfOneItem()
+    {
+        // Arrange
+        var dataRow = new DataRowAttribute(new[] { "a" }, new[] { "1" });
+        var methodInfoMock = new Mock<MethodInfo>();
+        methodInfoMock.SetupGet(x => x.Name).Returns("MyMethod");
+
+        // Act
+        var displayName = dataRow.GetDisplayName(methodInfoMock.Object, dataRow.Data);
+
+        // Assert
+        Verify(displayName == "MyMethod (System.String[],System.String[])");
```

Comment:
This display name and the next one are not ideal but we will address it in a different PR (see issue https://github.com/microsoft/testfx/issues/1333).